src/torrent.cpp

```diff
@@ -1456,130 +1522,80 @@ bool is_downloading_state(int const st)
 			if (!names.empty()) names += " | n: ";
 			names.append(torrent_name, name_length);
 #endif
-
 			if (std::strncmp(torrent_name, "*", name_length) == 0
-				|| std::strncmp(torrent_name, m_torrent_file->name().c_str(), name_length) == 0)
+				|| std::strncmp(torrent_name, expected.c_str(), name_length) == 0)
 			{
-#ifdef TORRENT_DISABLE_LOGGING
-				return true;
-#else
+#ifndef TORRENT_DISABLE_LOGGING
```

Comment:
this was inverted, I don't think that's necessary, or necessarily an improvement